bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusSenseGuardHandler.java

```diff
@@ -146,27 +184,38 @@ protected Data getLastDataPoint(Appliance appliance) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Could not load data from API.");
             return new Data();
         }
-        return applianceData.getData();
+        Data data = applianceData.getData();
+        Collections.sort(data.measurement, Comparator.comparing(e -> ZonedDateTime.parse(e.timestamp)));
+        Collections.sort(data.withdrawals, Comparator.comparing(e -> e.starttime));
+        return data;
     }
 
     private @Nullable ApplianceData getApplianceData(Appliance appliance) {
         Instant from = fromTime();
-        Instant to = Instant.now();
+        // Truncated to date only inside api package
+        Instant to = Instant.now().plus(1, ChronoUnit.DAYS);
 
         OndusService service = getOndusService();
         if (service == null) {
             return null;
         }
         try {
+            logger.debug("Fetching data for {} from {} to {}", thing.getUID(), from, to);
             BaseApplianceData applianceData = service.applianceData(appliance, from, to).orElse(null);
-            if (applianceData.getType() != Appliance.TYPE) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "Thing is not a GROHE SENSE Guard device.");
-                return null;
+            if (applianceData != null) {
+                if (applianceData.getType() != Appliance.TYPE) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Thing is not a GROHE SENSE Guard device.");
+                    return null;
+                }
+                return (ApplianceData) applianceData;
+            } else {
+                logger.debug("Could not load appliance data for thing {}", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to find applicance data");
```

Comment:
Typo: appliance. Please move text to src/main/resources/OH-INF/i18n/groheondus.properties to allow for localization.